### PR TITLE
Clarify explicit direction for renamed SQUASH_* variables

### PR DESCRIPTION
**Pull request by guidosarducci**
Based on discussion from the original commit 898846553f57dc8bd4be84fd8f3b5063f2b324c6.  This also anticipates adding/using "_EGRESS" versions of the variables.

### CODE REVIEW

File: src/defaults.sh [hunk 1]
```diff
@@ -19,8 +19,8 @@
 [ -z "$IECN" ] && IECN="ECN"
 [ -z "$EECN" ] && EECN="ECN"
 # These two used to be called something else; preserve backwards compatibility
-[ -z "$ZERO_DSCP" ] && ZERO_DSCP="${SQUASH_DSCP:-1}"
-[ -z "$IGNORE_DSCP" ] && IGNORE_DSCP="${SQUASH_INGRESS:-1}"
+[ -z "$ZERO_DSCP_INGRESS" ] && ZERO_DSCP_INGRESS="${SQUASH_DSCP:-1}"
+[ -z "$IGNORE_DSCP_INGRESS" ] && IGNORE_DSCP_INGRESS="${SQUASH_INGRESS:-1}"
```
Comment (tohojo) [inline src/defaults.sh]: Ah, but now there's no backwards compatibility with the previous change. The easiest is probably to keep the old variable for now and just do a conditional assignment to the _INGRESS version...

Comment (guidosarducci) [inline src/defaults.sh]: Ah right, was confused about backward compatibility until I realized you had already tagged and released the original commit. Thought I had a little window to fix things up... Updated my changes and pushing.

**Comment by guidosarducci** [comment]
@tohojo Thanks, would you mind a few follow-on questions?

Going forward, are you OK with using "WIP" type PRs to trial rationale and changes, and allow for some questions and discussion?  Could you also clarify a little the strategy for branching and tag/releases?  I'm asking to understand better how much of a window we have to make changes, or prepare coordinated/complementary PRs (hence the WIP question)?  For example, I saw a 1.1.x branch, thinking that was a stable release and target for backports, with new changes on master.

BTW, it appears the recent "cake wash" changes introduced a regression.  Previously, setting "SQUASH_DSCP" would **always** zero DSCP bits via ipt.  Now, for example in `layer_cake.qos`, iptables code was removed and all DSCP processing pushed into cake.  This has the side-effect that if shaping is disabled (e.g. `DOWNLINK == 0` which is common) then squashing of DSCP bits is bypassed.

Thanks again!

**Comment by tohojo** [comment]
guidosarducci <notifications@github.com> writes:

> Going forward, are you OK with using "WIP" type PRs to trial rationale
> and changes, and allow for some questions and discussion?

Sure, go right ahead :)

> Could you also clarify a little the strategy for branching and
> tag/releases? I'm asking to understand better how much of a window we
> have to make changes, or prepare coordinated/complementary PRs (hence
> the WIP question)? For example, I saw a 1.1.x branch, thinking that
> was a stable release and target for backports, with new changes on
> master.

Well, so far the release "strategy" (insofar such a thing has existed)
has been "tag a new release and update the openwrt package whenever we
have accumulated enough interesting changes that we think won't break
stuff".

I created the 1.1.x branch after LEDE forked off their release branch.
The idea for this is to introduce only bugfixes to that, and base LEDE
17.01 releases on that, then put all changes in functionality into the
main branch and release that occasionally with roughly the old
"strategy" :)

> BTW, it appears the recent "cake wash" changes introduced a
> regression. Previously, setting "SQUASH_DSCP" would always zero DSCP
> bits via ipt. Now, for example in layer_cake.qos, iptables code was
> removed and all DSCP processing pushed into cake. This has the
> side-effect that if shaping is disabled (e.g. DOWNLINK == 0 which is
> common) then squashing of DSCP bits is bypassed.

Ah, bugger. Is squashing DSCP needed if you're not shaping? Not sure
what people are actually using that for? WiFi?

-Toke

**Comment by moeller0** [comment]
Hi Toke,

> On Feb 5, 2017, at 21:04, Toke Høiland-Jørgensen <notifications@github.com> wrote:
> 
> guidosarducci <notifications@github.com> writes:
> 
> > Going forward, are you OK with using "WIP" type PRs to trial rationale
> > and changes, and allow for some questions and discussion?
> 
> Sure, go right ahead :)
> 
> > Could you also clarify a little the strategy for branching and
> > tag/releases? I'm asking to understand better how much of a window we
> > have to make changes, or prepare coordinated/complementary PRs (hence
> > the WIP question)? For example, I saw a 1.1.x branch, thinking that
> > was a stable release and target for backports, with new changes on
> > master.
> 
> Well, so far the release "strategy" (insofar such a thing has existed)
> has been "tag a new release and update the openwrt package whenever we
> have accumulated enough interesting changes that we think won't break
> stuff".
> 
> I created the 1.1.x branch after LEDE forked off their release branch.
> The idea for this is to introduce only bugfixes to that, and base LEDE
> 17.01 releases on that, then put all changes in functionality into the
> main branch and release that occasionally with roughly the old
> "strategy" :)
> 
> > BTW, it appears the recent "cake wash" changes introduced a
> > regression. Previously, setting "SQUASH_DSCP" would always zero DSCP
> > bits via ipt. Now, for example in layer_cake.qos, iptables code was
> > removed and all DSCP processing pushed into cake. This has the
> > side-effect that if shaping is disabled (e.g. DOWNLINK == 0 which is
> > common) then squashing of DSCP bits is bypassed.
> 
> Ah, bugger. Is squashing DSCP needed if you're not shaping? Not sure
> what people are actually using that for? WiFi?

	I agree that wifi with WMM might be affected. But I also believe that the set of people that used layer_cake/piece_of_cake with ingress set 0 that actively wanted dscp zeroing to be close to zero. We never advertised that as a feature. So I believer we should just live with that behavioral change until we get feed-back of users that are un-happy with the change, I believe it is not really a “regression” if nobody notices (and I believe the Guido found this by looking at the code not be seeing unexpected behaviour, otherwise that would count as the feed-back I was looking for ;))

Best Regards
	Sebastian

**Comment by guidosarducci** [comment]
@tohojo  Thanks for the explanation around branches/releases; I had gathered more or less what you outlined, and will also go down the WIP route for some coordinated changes I'm thinking about.
> Ah, bugger. Is squashing DSCP needed if you're not shaping? Not sure what people are actually using that for? WiFi?

Well, yes. And for the good network hygiene of shielding your internal networks from foreign diffserv.  Let's not forget also that squashing DSCP is a default setting, and advertised right on the GUI.  But the cake wash changes are also just part of a larger problem of inconsistent DSCP treatment across qdiscs and scripts. For example, right now squashing with/without shaping does work in `simple.qos`.  I expect (hope) these inconsistencies will be dealt with by some of the "WIP" changes I'm considering however.  In my mind, these sorts of issues complicate usage for normal users and maintainability of the SW.

@moeller0  I agree these bugs are not high-priority and don't warrant an urgent fix;  nonetheless, they should be resolved.  But suggesting there's no problem because no one noticed or cares, or dismissing feedback based on code review, makes for an implausible argument.  I just updated this very PR with backward-compatibility based on the near-zero chance someone used or noticed the prior SQUASH variables over the 2-3 days of their lifetime.  And the feedback from @tohojo prompting that change was based on (you guessed it) code review.

Thanks and take care,
Tony

**Comment by moeller0** [comment]
> On Feb 6, 2017, at 08:54, guidosarducci <notifications@github.com> wrote:
> @moeller0 I agree these bugs are not high-priority and don't warrant an urgent fix; nonetheless, they should be resolved. But suggesting there's no problem because no one noticed or cares, or dismissing feedback based on code review, makes for an implausible argument. 

	Then we have different definitions of implausible ;) . We can always change the luci field description to clarify that we only zero if ingress shaping is active. I actually consider us running ipt_setup unconditionally sub optimal and believe ipt_setup could well be separated into ingress and egress sections and only called if the respective shaper is activated. But I also believe that nobody is going to notice behaviorally anyways.
That said I agree that the remapping of ingress DSCPs would better live in its own section instead of living inside the Queuing Discipline tab, as you point out policy wise this can and should be decoupled from the actual implementation via iptables or cake’s internals.


> I just updated this very PR with backward-compatibility based on the near-zero chance someone used or noticed the prior SQUASH variables over the 2-3 days of their lifetime. And the feedback from @tohojo prompting that change was based on (you guessed it) code review.

Hey, I am not saying code review is bad, but that not all behavioral changes are regressions. Because let’s face it us changing layer_cake from diffserv4 to diffserv3 is a strong user observable change (not a critical bug fix) and we did that anyway. So we are not a strict no noticeable regressions allowed kind of project ;).

That said, if you I do not really object to fixing the zeroing issue, I just wanted to discuss whether it was noticed. (Not that my object would be relevant, this is after all more of a taste and style question than correctness.)

Best Regards
	Sebastian

**Pull request merged**